### PR TITLE
Implemented more packets and packet tests.

### PR DESCRIPTION
**Pull request by Seb-stian**
I implemented the rest of deserialization, including tests.

### CODE REVIEW

File: Obsidian/Serializer/Enums/DataType.cs [hunk 1]
```diff
@@ -2,7 +2,7 @@
 {
     public enum DataType
     {
-        Auto = 0,
+        Auto,
```
Comment (Craftplacer) [inline Obsidian/Serializer/Enums/DataType.cs:5]: Was the null redundant?

File: Obsidian/Serializer/Enums/DataType.cs [hunk 2]
```diff
@@ -21,10 +21,14 @@ public enum DataType
         Slot,
         NbtTag,
         Position,
+        AbsolutePosition,
```
Comment (Craftplacer) [inline Obsidian/Serializer/Enums/DataType.cs:30]: Mind telling what's the difference between Position and AbsolutePosition?

**Review by Craftplacer — APPROVED** [review]

File: Obsidian/Serializer/Enums/DataType.cs (hunk 2 above)
Comment (Tides) [reply Obsidian/Serializer/Enums/DataType.cs:30]: Absolute position uses the Position class but se/de-serializes by reading/writing the x, y and z separately while Position would usually mean you're getting/writing the values from/to a long

Comment (Craftplacer) [reply Obsidian/Serializer/Enums/DataType.cs:30]: Hmm, weird name choice.
Might be worth documenting this encoding on the enum value?

Comment (Tides) [reply Obsidian/Serializer/Enums/DataType.cs:30]: Sounds good, lets do that

**Review by Tides — APPROVED** [review]

File: Obsidian/Serializer/Enums/DataType.cs (hunk 1 above)
Comment (Seb-stian) [reply Obsidian/Serializer/Enums/DataType.cs:5]: The 0 was redundant, yes.

**Pull request merged**